Azure/azure-powershell
# Test Importing Az.Accounts in Parallel

## Description

Add a case to smoke test to import Az.Accounts in parallel PowerShell processes.

Because 'Start-Job' is not supported by design in scenarios where PowerShell is being hosted in other applications, for example ADO pipeline, I skipped the test case by checking if [an environment variable defined only in ADO](https://docs.microsoft.com/en-us/azure/devops/pipelines/build/variables?view=azure-devops&tabs=yaml) exists or not.

Although in ADO it is skipped, it can still run locally or on cloud shell, so it's still meaningful.

### How to Test

1. I test the official Az v8.0.0 release. Due to the file locking issue (#18321) the run failed as expected.
```
\code\azure-powershell\tools\Test\SmokeTest\RmCoreSmokeTests.ps1 : System.Management.Automation.RuntimeException: Failed to invoke script
of Import Az.Accounts in Parallel. Some jobs have failed.
At line:1 char:1
+ .\code\azure-powershell\tools\Test\SmokeTest\RmCoreSmokeTests.ps1
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
    + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,RmCoreSmokeTests.ps1
```
1. I made a signed build based on #18483 and ran the smoke test. The issue is fixed.



## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
